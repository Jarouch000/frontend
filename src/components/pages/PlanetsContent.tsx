import React from "react";

import {Planets} from '../Planets';
import Links from '../../links';

/**
 * Represents content with an interactive demo of planetary and satellites datasets.
 */
const PlanetsContent = () => {
  return (
    <div>

      <div className='contentCenter'>
        This is an interactive version of <a href='https://nssdc.gsfc.nasa.gov/planetary/factsheet/'>Planetary Fact
        Sheet</a> and <a href='https://ssd.jpl.nasa.gov/?sat_phys_par'>Planetary Satellite Physical Parameters</a>
        &nbsp;datasets from {Links.jpl}.
        <p>
          Tips: <span className='highlight'>Select the planet below</span> to see only its satellites
          or <span className='highlight'>click on the column</span> to sort.
        </p>
      </div>

      <hr/>

      <div className='planets'>
        <Planets/>
      </div>

    </div>
  );
};


export default PlanetsContent;
