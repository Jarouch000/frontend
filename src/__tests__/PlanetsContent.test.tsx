import React from 'react';
import { shallow } from 'enzyme';
import PlanetsContent from '../components/pages/PlanetsContent';
describe('PlanetsContent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<PlanetsContent />);

        expect(component).toMatchSnapshot();
    });
});
