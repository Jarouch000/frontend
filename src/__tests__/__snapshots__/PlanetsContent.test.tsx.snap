// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetsContent should render correctly in "debug" mode 1`] = `
<div>
  <div
    className="contentCenter"
  >
    This is an interactive version of 
    <a
      href="https://nssdc.gsfc.nasa.gov/planetary/factsheet/"
    >
      Planetary Fact Sheet
    </a>
     and 
    <a
      href="https://ssd.jpl.nasa.gov/?sat_phys_par"
    >
      Planetary Satellite Physical Parameters
    </a>
    Â datasets from 
    <a
      href="https://www.jpl.nasa.gov/"
    >
      NASA Jet Propulsion Laboratory
    </a>
    .
    <p>
      Tips: 
      <span
        className="highlight"
      >
        Select the planet below
      </span>
       to see only its satellites or 
      <span
        className="highlight"
      >
        click on the column
      </span>
       to sort.
    </p>
  </div>
  <hr />
  <Connect(Planets) />
</div>
`;
